ts/Dashboards/Plugins/DataGridComponent.ts

```diff
@@ -378,6 +381,39 @@ class DataGridComponent extends Component {
         }
     }
 
+    /**
+     * Based on the `columnsToShow` option, filter the columns of the table.
+     *
+     * @internal
+     */
+    private filterColumns(): DataTable|undefined {
+        const table = this.connector?.table.modified,
+            columnsToShow = this.options.columnsToShow;
+
+        if (table) {
+            const columnsToDelete = table.getColumnNames().filter((columnName): boolean => {
+                if (columnsToShow?.length) {
```

Comment:
The method can probably just return `table` earlier `if( !columnsToShow.length )`.